### ISSUE TITLE
Make S3Client take multiple implementations for integration tests (docker)

### ISSUE THREAD

**Issue by max8github** [akka/alpakka#675]
There seem to be currently no way to set an arbitrary endpoint URL with `S3Client`.

That would be useful for example for being able to run integration tests with something like `LocalStack`, which can mimic Amazon S3 with a local docker deployment.

The code that sets the URL for S3 seems to be in (i'm looking at master):
```
akka.stream.alpakka.s3.impl.HttpRequests.requestHost()
```
which is used by `HttpRequests.s3Request()`, which ends up being used by all important `S3Client` methods through `S3Stream`.

I attempted to write a patch around adding or allowing this functionality, but i realized i first needed some direction/discussion first no how you would go about it. Could you please provide that?

At the moment, i generate a customized "internal" jar to be deployed in our company's Nexus server (it is a 15 minutes hack to your code, which i don't dare to show). Of course, would be nice if `S3Client` allowed more flexible settings or implementations to avoid that.

For example, with `com.amazonaws.services.s3.AmazonS3` it is possible to inject a different implementation with, say, something like:
```
    val endpointConfiguration = new AwsClientBuilder.EndpointConfiguration(url, region.getName)
    AmazonS3ClientBuilder.standard()
      .enablePathStyleAccess()
      .withEndpointConfiguration(endpointConfiguration)
      .withCredentials(serviceConfig.s3)
      .build()
```

**Comment by max8github** [comment]
Realizing that this issue is resolved by issue #677 , closing.

**Comment by ennru** [comment]
I'll press the close button, cheers.
